src/main/java/life/qbic/data/processing/registration/ProcessRegistrationRequest.java

```diff
@@ -40,53 +52,134 @@ public class ProcessRegistrationRequest extends Thread {
   private final ConcurrentRegistrationQueue registrationQueue;
   private final Path workingDirectory;
   private final Path targetDirectory;
+  private final String metadataFileName;
+  private final Path usersErrorDirectory;
   private AtomicBoolean active = new AtomicBoolean(false);
 
   public ProcessRegistrationRequest(@NonNull ConcurrentRegistrationQueue registrationQueue,
-      @NonNull RegistrationConfiguration configuration) {
+      @NonNull RegistrationConfiguration configuration, @NonNull GlobalConfig globalConfig) {
     this.setName(threadName.formatted(nextThreadNumber()));
     this.registrationQueue = registrationQueue;
     this.workingDirectory = configuration.workingDirectory();
     this.targetDirectory = configuration.targetDirectory();
+    this.metadataFileName = configuration.metadataFileName();
+    this.usersErrorDirectory = globalConfig.usersErrorDirectory();
   }
 
   private static int nextThreadNumber() {
     return threadNumber++;
   }
 
-  @Override
-  public void run() {
-    while (true) {
-      var request = registrationQueue.poll();
-      active.set(true);
-      log.info("Processing request: {}", request);
-      try {
-        Path taskDir = createTaskDirectory();
-        Path newLocation = taskDir.resolve(request.target().getFileName());
-        Files.move(request.target(), newLocation);
-        writeProvenanceInformation(taskDir, newLocation, request);
-        Files.move(taskDir, targetDirectory.resolve(taskDir.getFileName()));
-      } catch (RuntimeException e) {
-        log.error("Error moving task directory", e);
-        // TODO move back to user folder
-      } catch (IOException e) {
-        log.error("Error while processing registration request", e);
-        // TODO move back to user folder
-      } finally {
-        active.set(false);
-        log.info("Processing completed: {}", request);
+  private static void cleanup(Path workingTargetDir) throws IOException {
+    try (var content = Files.walk(workingTargetDir)) {
+      content.map(Path::toFile).sorted(Comparator.reverseOrder()).forEach(File::delete);
+    }
+  }
+
+  private static void processMeasurement(String measurementId,
+      Map<String, List<RegistrationMetadata>> aggregatedFilesByMeasurementId, Path workingTargetDir,
+      Path taskDir) throws IOException {
+    for (RegistrationMetadata metadataEntry : aggregatedFilesByMeasurementId.get(
+        measurementId)) {
+      Files.move(workingTargetDir.resolve(metadataEntry.file()),
+          taskDir.resolve(metadataEntry.file()));
+    }
+  }
+
+  private void moveBackToOrigin(Path target, Path usersHomePath, String reason) {
+    log.info("Moving back to original user directory: {}", usersHomePath);
+    try {
+      Path taskDir = createTaskDirectory();
+      Files.move(target, taskDir.resolve(target.getFileName()));
+      var errorFile = taskDir.resolve("error.txt").toFile();
+      errorFile.createNewFile();
```

Comment:
Does this mean that the errorfile is overwritten for each measurement if the user tries to register multiple faulty measurements?